### DISCUSSION TITLE
Difference between "clear_terminal scroll" and "clear_terminal to_cursor_scroll"

### DISCUSSION THREAD

**Discussion by rafalglowacz** [kovidgoyal/kitty#7214] [Q&A]
Thanks for another Kitty release! I'm looking at the change log and I have a hard time understanding the new mode for `clear_terminal` called `to_cursor_scroll`. What's the difference between it and the old `scroll` mode?

`scroll` has been my choice ever since I learned about it so I'm wondering if the new mode is some kind of an upgrade for it. But I don't see a difference between them. I have a two line prompt and they both handle it the same way. I tried to see what'll happen when using `less` since then the cursor isn't at the top, but the behavior also seems identical. So, what's the difference? :slightly_smiling_face:

**Comment by kovidgoyal** [comment]
to_cursor_scroll scrolls only upto the line with the cursor or the first
line of the prompt when shell integration is active and the cursor is at
a prompt. scroll, scrolls everything including the prompt. The shell
then has to redraw the prompt. Therefore, the former works usually without
co-operation from the shell, the latter does not.
